datadog/fakeintake/ecsFargate.go

```diff
@@ -0,0 +1,89 @@
+package fakeintake
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/aws"
+	tidECS "github.com/DataDog/test-infra-definitions/aws/ecs"
+	"github.com/cenkalti/backoff/v4"
+
+	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/awsx"
+	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/ecs"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	containerName = "fakeintake"
+	port          = 80
+	oneSecond     = 1000
+	sleepInterval = 1 * oneSecond
+	maxRetries    = 120
+)
+
+func ECSFargateService(e aws.Environment, name string, clusterArn string) (ipAddress pulumi.StringOutput, err error) {
+	taskDef, err := fargateLinuxTaskDefinition(e, name)
+	if err != nil {
+		return pulumi.StringOutput{}, err
+	}
+	fargateService, err := tidECS.FargateService(e, e.Namer.ResourceName(name), pulumi.String(clusterArn), taskDef.TaskDefinition.Arn())
+
+	// Hack passing taskDef.TaskDefinition.Arn() to execute apply function
+	// when taskDef has an ARN, thus it is defined on AWS side
+	ipAddress = pulumi.All(taskDef.TaskDefinition.Arn(), fargateService.Service.Name()).ApplyT(func(args []any) (string, error) {
+		var ipAddress string
+		err := backoff.Retry(func() error {
+			fmt.Println("waiting for task private ip")
+			serviceName := args[1].(string)
+			ecsClient, err := tidECS.NewECSClient(e.Ctx.Context(), e.Region())
+			if err != nil {
+				return err
+			}
+			ipAddress, err = ecsClient.GetTaskPrivateIP(clusterArn, serviceName)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("task private ip found: %s\n", ipAddress)
+			return err
+		}, backoff.WithMaxRetries(backoff.NewConstantBackOff(sleepInterval), maxRetries))
+		return ipAddress, err
+	}).(pulumi.StringOutput)
+
+	return ipAddress, err
+}
+
+func fargateLinuxTaskDefinition(e aws.Environment, name string) (*ecs.FargateTaskDefinition, error) {
+	return ecs.NewFargateTaskDefinition(e.Ctx, e.Namer.ResourceName(name), &ecs.FargateTaskDefinitionArgs{
+		Containers: map[string]ecs.TaskDefinitionContainerDefinitionArgs{
+			containerName: *fargateLinuxContainerDefinition(),
+		},
+		Cpu:    pulumi.StringPtr("256"),
+		Memory: pulumi.StringPtr("512"),
+		ExecutionRole: &awsx.DefaultRoleWithPolicyArgs{
+			RoleArn: pulumi.StringPtr(e.ECSTaskExecutionRole()),
+		},
+		TaskRole: &awsx.DefaultRoleWithPolicyArgs{
+			RoleArn: pulumi.StringPtr(e.ECSTaskRole()),
+		},
+		Family: e.CommonNamer.DisplayName(pulumi.String("fakeintake-ecs")),
+	}, e.ResourceProvidersOption())
+}
+
+func fargateLinuxContainerDefinition() *ecs.TaskDefinitionContainerDefinitionArgs {
+	return &ecs.TaskDefinitionContainerDefinitionArgs{
+		Cpu:         pulumi.IntPtr(100),
```

Comment:
Misleading, container-level limit are not applied unless there's CPU contention. As this is the only container in the Task, it will use everything available (up to task cpu limit).

Reply:
So I should omit them ?

Reply:
You can omit Memory, I think you need to explicitly set CPU to `0` as the Pulumi provider mandates it.

Reply:
I see they're required in the `TaskDefinition`, not int the `TaskDefinitionContainerDefinition`, removing this